sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -434,6 +434,33 @@ public TableReference apply(String jobUuid) {
   }
 
   /**
+   * A policy for determining output table names. Currently supports examining the elemented or
+   * the window.
+   */
+   public abstract static class TableWritePolicy<T> implements SerializableFunction<
+      TableWritePolicy.Context, T> {
```

Comment:
Rename T to TableT or something like that? For a while I thought it refers to the type of the element.